feat: pokemon_snackbar

Miqqy0069 opened the pull request:
- added a snackbar for the error of the Loading Action                                                    
- changed error message into No pokemons available

(working app)

https://user-images.githubusercontent.com/75653480/227883578-744023d0-f82d-4d35-93cc-dfdf140dc89c.mp4

(analyze) 
![image_360](https://user-images.githubusercontent.com/75653480/227873324-81a30a7a-a968-44a9-b348-b90f6159d3c7.png)

(dart format)
![image_720](https://user-images.githubusercontent.com/75653480/227873333-add08faf-e30f-4d40-b4d2-060ac6e61315.png)

Miqqy0069 committed:
feat: union sealed class
>> - added a snackbar for the error of the Loading Action                                                    - changed error message into No pokemons available

Miqqy0069 committed:
feat: pokemon snack bar
- dart format to 120

Miqqy0069 committed:
feat: pokemon snackbar
- added a private method for the snack bar
- changed error message for overview page and snack bar

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -18,13 +18,22 @@ class PokemonOverviewPage extends StatelessWidget {
+          );
+          return Center(
+            child: Text(
+              errorMessage ?? emptyString,
```
instead of errorMessage, use your `pokemonOverviewPageErrorMessage` constant instead 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -18,13 +18,22 @@ class PokemonOverviewPage extends StatelessWidget {
-          ),
-        ),
+        error: (errorMessage) {
+          _noPokemonAvailableDialog(
```
no need for extra trailing commas here 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +50,16 @@ class PokemonOverviewPage extends StatelessWidget {
     );
   }
+
+  void _noPokemonAvailableDialog(BuildContext context, errorMessage) {
```
kindly wrap this snackbar implementation in a separate function and name it with `_showErrorMessageSnackbar` 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +50,16 @@ class PokemonOverviewPage extends StatelessWidget {
+  void _noPokemonAvailableDialog(BuildContext context, errorMessage) {
+    const snackBar = SnackBar(
+      content: Text(
+        snackBarErrorMessage,
```
why not use `errorMessage` parameter instead? 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +50,16 @@ class PokemonOverviewPage extends StatelessWidget {
+        style: TextStyle(color: Colors.red),
+      ),
+    );
+    WidgetsBinding.instance.addPostFrameCallback((_) {
```
do you still need this addPostFrameCallback, seems not anymore since context is already available here as given from this build method 🙂

Miqqy0069 replied:
I've tried to not include this, but it always gives me an error that the snackbar can't be called during the build, so just to give it a delay when changing state po. and also my material app is on connector. so I think whenever i remove this, it always looks for a scaffold descendants.

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_vm.dart:
```diff
@@ -12,7 +12,7 @@ class PokemonOverviewVmFactory extends VmFactory<AppState, PokemonOverviewConnec
   Async<List<Pokemon>> _pokemons() {
     if (state.wait.isWaitingFor(GetPokemonsAction.key)) return const Async.loading();
-    if (state.pokemons.isEmpty) return const Async.error(errorMessageText);
+    if (state.pokemons.isEmpty) return const Async.error(pokemonOverviewPageErrorMessage);
```
isn't it you're supposed to pass `snackBarErrorMessage` instead? 🤔

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Miqqy0069 committed:
feat: pokemon snackbar
pr feedback
- removed private function for snackbar
- placed snackbar under error block

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -18,13 +18,27 @@ class PokemonOverviewPage extends StatelessWidget {
-          ),
-        ),
+        error: (errorMessage) {
+          final snackBar = SnackBar(
```
see cleo's pr for reference on how she segregated the snackbar implementation with the postFrameCallback 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -18,13 +18,27 @@ class PokemonOverviewPage extends StatelessWidget {
+          });
+          return const Center(
+            child: Text(
+              pokemonOverviewPageErrorMessage ?? emptyString,
```
pokemonOverviewPageErrorMessage constant has no way it's gonna be null right? therefore do u still need to perform null coalesce here? 😉

shawn-ffufm commented on lib/utilities/constants.dart:
```diff
@@ -8,4 +8,5 @@ const pokedexTitle = 'Pokedex';
 const indexKey = '[index]';
 
-const errorMessageText = 'Something went wrong';
+const snackBarErrorMessage = 'Something went wrong!';
```
errorMessage is enough for its name here, remember this message won't only be usable for snackbar it could be in any other case as well 🙂

shawn-ffufm commented on lib/utilities/constants.dart:
```diff
@@ -8,4 +8,5 @@ const pokedexTitle = 'Pokedex';
 
-const errorMessageText = 'Something went wrong';
+const snackBarErrorMessage = 'Something went wrong!';
+const pokemonOverviewPageErrorMessage = 'No Pokemons Available';
```
this isn't an error message right, rather a label? kindly rename this to `noPokemonsAvailableLabel` instead 🙂

shawn-ffufm reviewed (changes requested):
see minor pr feedback 🙂

Miqqy0069 committed:
feat: pokemon snackbar
- pr feedback
- renamed error message to label
- segregated the snackbar implementation with the postFrameCallback

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +47,15 @@ class PokemonOverviewPage extends StatelessWidget {
     );
   }
+
+  _showErrorMessageSnackBar(context, errorMessage) {
```
where's the function type here? this should be void right?

shawn-ffufm replied:
also, where are the parameter types as well? 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +47,15 @@ class PokemonOverviewPage extends StatelessWidget {
+        content: Text(
+      errorMessage ?? emptyString,
+      style: const TextStyle(
+        color: Colors.red,
```
no need for extra trailing comma

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -41,4 +47,15 @@ class PokemonOverviewPage extends StatelessWidget {
+      style: const TextStyle(
+        color: Colors.red,
+      ),
+    ));
```
add extra trailing comma here 🙂

shawn-ffufm reviewed (changes requested):
see minor pr feedback 🙂

Miqqy0069 committed:
feat: pokemon snackbar
- removed unnecesarry trailing commas
- added needed trailing commas
- added void function for the Snackbar
- added parameter types for the Snackbar function

shawn-ffufm reviewed (approved):
lgtm
